src/Microsoft.ML.TimeSeries/SrCnnEntireTransformer.cs

```diff
@@ -0,0 +1,903 @@
+            }
+            else
+            {
+                OutputLength = 0;
+            }
+
+            AnomalyModel = new SrCnnEntireModeler(Host, this, input);
+        }
+
+        private SrCnnEntireTransformer(IHost host, ModelLoadContext ctx)
+            : base(host, ctx)
+        {
+            InputColumnName = ctx.Reader.ReadString();
+            Host.CheckDecode(!String.IsNullOrEmpty(InputColumnName));
+
+            OutputColumnName = ctx.Reader.ReadString();
+            Host.CheckDecode(!String.IsNullOrEmpty(OutputColumnName));
+
+            Threshold = ctx.Reader.ReadDouble();
+            Host.CheckDecode(Threshold >= 0 && Threshold <= 1);
+
+            BatchSize = ctx.Reader.ReadInt32();
+            Host.CheckDecode(BatchSize == -1 || BatchSize >= 12);
+
+            SrCnnDetectMode = (SrCnnDetectMode)ctx.Reader.ReadInt32();
+
+            Sensitivity = ctx.Reader.ReadDouble();
+            Host.CheckDecode(Sensitivity >= 0 && Sensitivity <= 100);
+
+            OutputLength = ctx.Reader.ReadInt32();
+            Host.CheckDecode(OutputLength == 3 || OutputLength == 4 || OutputLength == 7);
+
+            ctx.LoadModel<SrCnnEntireModeler, SignatureLoadModel>(host, out AnomalyModel, "SrCnnEntire");
+            Host.CheckDecode(AnomalyModel != null);
+        }
+
+        private static IDataTransform Create(IHostEnvironment env, Options options, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(options, nameof(options));
+            env.CheckValue(input, nameof(input));
+
+            return new SrCnnEntireTransformer(env, options, input).MakeDataTransform(input);
+        }
+
+        internal static SrCnnEntireTransformer Create(IHostEnvironment env, ModelLoadContext ctx)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register(LoaderSignature);
+
+            host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+
+            return new SrCnnEntireTransformer(host, ctx);
+        }
+
+        private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, DataViewSchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
+        private protected override void SaveModel(ModelSaveContext ctx)
+        {
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            Host.Assert(Threshold >= 0 && Threshold <= 1);
+            Host.Assert(BatchSize == -1 || BatchSize >= 12);
+            Host.Assert(Sensitivity >= 0 && Sensitivity <= 100);
+            Host.Assert(OutputLength == 3 || OutputLength == 4 || OutputLength == 7);
+
+            base.SaveColumns(ctx);
+            ctx.Writer.Write(InputColumnName);
+            ctx.Writer.Write(OutputColumnName);
+            ctx.Writer.Write(Threshold);
+            ctx.Writer.Write((Int32)BatchSize);
```

Comment:
>Int32 [](start = 30, length = 5)

`BatchSize` is an int, no need to cast.

Reply:
Same for `OutputLength`.

---
In reply to: [415779054](https://github.com/dotnet/machinelearning/pull/4998#discussion_r415779054) [](ancestors = 415779054)

Reply:
Updated.